goyaml.v3/encode_test.go

```diff
@@ -731,6 +731,40 @@ func (s *S) TestSortedOutput(c *C) {
 	}
 }
 
+func (s *S) TestQuotingMaintained(c *C) {
+	var buf bytes.Buffer
+	var yamlValue map[string]interface{}
+	const originalYaml = `data:
+    A1: "0x0000000000000000000000010000000000000000"
+    A2: "0x000000000000000000000000FFFFFFFFFFFFFFFF"
+    A3: "1234"
+    A4: 0x0000000000000000000000010000000000000000
+    A5: 0x000000000000000000000000FFFFFFFFFFFFFFFF
```

Comment:
This repo is oriented at yaml which can be round-tripped to json and used in Kubernetes API objects

These bigints would be problematic for Kubernetes as they do not round-trip to json ... I'm on the fence that we should enable this in this library

Reply:
My general thought process was that unquoted values are already not round-trippable to JSON when bigger than 64 bytes, but if quoted those should be maintained. The current implementation assumes that if an integer value does not fit a 64-byte integer then it must be a string but that's not exactly true.